### PR TITLE
Add unit tests for tokenizers and filters

### PR DESCRIPTION
**Pull request by mocobeta**
Some tokenizers/filters seem to have no unit test (e.g. SimpleTokenizer and StopWordFilter).
I think it would be nice to add basic tests for them for future development. To start with, I added a test module for SimpleTokenizer; does it make sense?

### CODE REVIEW

**Comment by PSeitz** [comment]
@mocobeta That's a really good idea to cover these with tests

**Comment by mocobeta** [comment]
@PSeitz thanks for your reply. I'll try to add tests for other components.

**Comment by fulmicoton** [comment]
@mocobeta awesome. I'm happy to merge this, but the PR is in draft state.

**Comment by mocobeta** [comment]
Thanks, @fulmicoton. I'd like to include a few more tests for other tokenizers, and then I will open this soon.

File: src/tokenizer/lower_caser.rs [hunk 1]
```diff
@@ -56,31 +56,30 @@ impl<'a> TokenStream for LowerCaserTokenStream<'a> {
...
+    use crate::tokenizer::tests::assert_token;
+    use crate::tokenizer::{LowerCaser, SimpleTokenizer, TextAnalyzer, Token};
 
     #[test]
     fn test_to_lower_case() {
-        assert_eq!(
-            lowercase_helper("Русский текст"),
-            vec!["русский".to_string(), "текст".to_string()]
-        );
+        let tokens = token_stream_helper("Tree");
```
Comment (mocobeta) [inline src/tokenizer/lower_caser.rs:64]: I refactored LowerCaser test so that positions and offsets are also verified.

**Comment by mocobeta** [comment]
I think this covers all existing tokenizers/filters. Tests added here are very basic ones though, I hope this will be of some help.

**Comment by fulmicoton** [comment]
@mocobeta It definitely helps! Thank you!

**Pull request merged**